### PR TITLE
add crypto build to CI

### PR DESCRIPTION
**Pull request by koko1123**
## Goal of this PR
Fix persistent crypto build issue in CI

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -71,9 +71,7 @@ jobs:
       # In order to be able to build with flow-go and the relic tag, we need to
       # run its go generate target.
       - name: Make Flow Go's crypto
-        run: |
-          cd ./flow-go
-          make crypto_setup_gopath
+        run: make crypto_setup
```
Comment (peterargue) [inline .github/workflows/ci.yml:74]: isn't this what `make crypto_setup_gopath` is already doing? Maybe @tarakby has some ideas for how to make this more reliable

Comment (tarakby) [reply .github/workflows/ci.yml:74]: What I see now is the right way to import flow-go/crypto in a project, as described in the package Readme :
 - we can delete the commented line https://github.com/onflow/flow-archive/blob/master/go.mod#L278
 - what is the reason of deleting the line https://github.com/onflow/flow-archive/blob/master/crypto_build.sh#L18. It actually allows the crypto version to be synched with what's imported `go.mod` without entering it [manually](https://github.com/onflow/flow-archive/blob/master/crypto_build.sh#L6).
 
 
`crypto_setup_gopath` is a make target that is defined by the `flow-go` repo, but this is a different repo, so it doesn't have to use the same target (flow-go/crypto is an independent package, it doesn't require flow-go to work). 
 This said, what is the purpose of checking out `onflow/flow-go` in the CI job above?
 If it was just to run `crypto_setup_gopath`, we should be able to delete it.

**Review by zhangchiqing — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -13,60 +13,61 @@ require (
...
+	golang.org/x/sync v0.1.0
+	google.golang.org/api v0.114.0
+	google.golang.org/grpc v1.53.0
 )
 
 require (
-	github.com/onflow/flow-go v0.29.17-read-leaf-nodes
-	github.com/onflow/flow-go/crypto v0.24.4
-	github.com/optakt/flow-dps-rosetta v1.0.0
+	github.com/onflow/flow-go v0.30.1
```
Comment (zhangchiqing) [inline go.mod:30]: we need to upgrade to `v0.30.3-fix-read-leaf-nodes`, as [this PR did](https://github.com/onflow/flow-archive/pull/96/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R30)

File: service/mapper/transitions.go [hunk 3]
```diff
@@ -127,12 +127,18 @@ func (t *Transitions) BootstrapState(s *State) error {
 
 	log.Info().Msgf("bootstrap with checkpoint file %v%v", s.checkpointDir, s.checkpointFileName)
 
+	// reading the leaf nodes on a goroutine
+	// and use a doneRead channel to report if running into any error
+	doneRead := make(chan error, 1)
```
[the remark this answers is not in the document]
Comment (zhangchiqing) [reply service/mapper/transitions.go:132]: Let's reverse all the changes from this file

**Review by SaveTheRbtz — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (koko1123) [reply .github/workflows/ci.yml:59]: Done!

**Pull request merged**